### DISCUSSION TITLE
Can LEAP be compared with other Backprojection Methods?

### DISCUSSION THREAD

**Discussion by ProjX0** [llnl/LEAP#47] [General]
I understand that LEAP uses the Separable Footprint (SF-RR) method during the Backprojection process.

Would it be possible to compare LEAP with Ray-driven, Pixel-driven, and Distance-driven methods?

Thank you.

**Comment by kylechampley** [comment]
The projectors in LEAP are very similar to SF-RR, but they are not the same.  There are some subtle differences.

When Hyojin and I started writing LEAP, our goal was to provide matched Siddon, Joseph, and Separable Footprint projectors and all CT geometries.  If you look at the earlier commits, you will see a function called set_projector() which allows users to switch between various models.  We depreciated this feature because it was too hard to maintain so many projectors and the matched Siddon projectors gave terrible results.  The matched Joseph projectors work OK, but once we figured out how to speed up the SF projectors they were as fast as the other projectors and provided better results, so we just stopped supporting the other projector types.

So currently you cannot compare different projectors with LEAP.

There are papers that compare projectors such as the papers by Basu and DeMan referenced on the LEAP main page and Yong Long's SF paper.

**Comment by kylechampley** [comment]
I actually ended up adding voxel-driven backprojection to LEAP in the latest release.  Analysis of some of the differences can be found [here](https://github.com/LLNL/LEAP/blob/main/results/SF_vs_VD.md).

**Reply by ProjX0** [reply]
Thanks, Kyle! I'll check it out when I get a chance.
